docs/howto/partitions.rst

```diff
@@ -0,0 +1,167 @@
+   application will use partitions:
+
+   .. code-block:: python
+
+     from craft_parts import Features
+
+     Features.reset()
+     Features(enable_partitions=True)
+
+   .. NOTE::
+      The ``craft-application`` class :class:`AppFeatures
+      <craft_application.AppFeatures>` has a similar name and serves a similar
+      purpose to ``craft-parts``'s :class:`Features <craft_parts.Features>`,
+      but partitions cannot be enabled via :class:`AppFeatures
+      <craft_application.AppFeatures>`!
+
+#. Define the list of partitions
+
+   We need to tell the :class:`LifecycleManager <craft_parts.LifecycleManager>`
+   class about our partitions, but applications do not usually directly
+   instantiate the LifecycleManager.
+
+   Instead, override your :class:`Application
+   <craft_application.Application>`'s ``_setup_partitions`` method, and return
+   a list of the partitions, which will eventually be passed to the
+   :class:`LifecycleManager <craft_parts.LifecycleManager>`:
+
+   .. code-block:: python
+
+     class SnackcraftApplication(Application):
+
+       ...
+
+       @override
+       def _setup_partitions(self, yaml_data: dict[str, Any]) -> list[str] | None:
+           return ["default", "kernel", "component/bar-baz"]
+
+Using the partitions
+====================
+
+Partitions cannot be used until `after you have configured your application
+<#app-changes>`_.
+
+In configuration yaml
+---------------------
+
+Defined partitions may be referenced in the ``organize``, ``stage``, and
+``prime`` sections of your configuration yaml files:
+
+.. code-block:: yaml
+
+  organize:
+    <source-path>: (<partition>)/<path>
+  stage:
+    - (<partition>)/<path>
+  prime:
+    - (<partition>)/<path>
+
+Paths in the configuration yaml not beginning with a partition label will
+implicitly use the default partition.
+
+The source path of an ``organize`` entry can only be from the default
+partition.  For example:
+
+.. code-block:: yaml
+
+  organize:
+    (kernel)/usr/local/bin/hello: bin/hello
+
+.. code-block:: text
+
+  Cannot organize files from 'kernel' partition.
+  Files can only be organized from the 'default' partition
+
+When the ``stage`` and ``prime`` keywords are not provided for a part,
+craft-parts' default behavior is to stage and prime all files for the part in
+all partitions.
+
+(If a stage or prime filter *is* applied to a partition, the default behavior
+will not be affected for the other partitions.)
+
+See also
+^^^^^^^^
+
+ * Craft parts: part properties: `organize`_
+ * Craft parts: filesets: `specifying paths`_
+
+In environment variables
+------------------------
+
+You might use these variables in a lifecycle override section of a
+configuration yaml.  For instance:
+
+.. code-block:: yaml
+
+  override-prime: |
+    cp -R $CRAFT_KERNEL_STAGE/vmlinux $CRAFT_KERNEL_PRIME/
+    chmod -R 444 $CRAFT_KERNEL_PRIME/*
+    cp -R $CRAFT_STAGE/lib/modules/6.x/* $CRAFT_PRIME
+    chmod -R 600 $CRAFT_PRIME/*
```

Comment:
This might not be the best example because it's migrating from stage to prime by hand and moving files around manually is not a recommended practice (it may work in some cases, but it can also result in inconsistent state files). Just removing the origin variable in the example would fix this.

Reply:
Is this correct? https://github.com/canonical/craft-application/pull/371/commits/0b09bae8242a3208eaa16eec77f6162c592875a4